core/src/main/scala/chisel3/Data.scala

```diff
@@ -324,8 +324,13 @@ trait BaseType extends HasId with NamedComponent {
   private[chisel3] def specifiedDirection: SpecifiedDirection = _specifiedDirection
   private[chisel3] def specifiedDirection_=(direction: SpecifiedDirection) = {
     _specifiedDirection = direction
+    _passiveDirection = direction
   }
 
+  // Passive direction for this node, ignoring coercion from Input and Output.
+  private var _passiveDirection:         SpecifiedDirection = SpecifiedDirection.Unspecified
+  private[chisel3] def passiveDirection: SpecifiedDirection = _passiveDirection
+
```

Comment:
I don't think we need this var on all Data because Elements always have passive direction. Considering a var is an extra 4-bytes of memory use, this needs to only be added to `Record` (Vecs can just check their `sample_element`). 

You can have the def on `Data` and then implement it as `true` on `Element` and do the implementation with a `var` on `Record`.

Reply:
With this suggestion you'll want to set `_passiveDirection` during `Record.bind`.